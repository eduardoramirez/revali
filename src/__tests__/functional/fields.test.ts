import {getIntrospectionQuery, graphql, printSchema} from 'graphql'

import {compileSchema, Field, list, nullable, ObjectType, TSGraphQLString} from 'revali/index'

describe('Fields', () => {
  it('generates a schema for valid fields', async () => {
    @ObjectType()
    class SampleNestedObject {
      @Field()
      public stringField!: string
    }

    @ObjectType()
    class SampleObject {
      @Field()
      public implicitStringField!: string

      @Field({type: SampleNestedObject})
      public objectField!: SampleNestedObject

      @Field({type: nullable(TSGraphQLString)})
      public explicitNullableStringField: string | undefined

      @Field({type: list(TSGraphQLString)})
      public implicitStringArrayField!: string[]

      @Field({type: nullable(list(TSGraphQLString))})
      public nullableArrayFieldNew: string[] | undefined

      @Field({type: nullable(list(SampleNestedObject))})
      public nullableObjectArrayField: SampleNestedObject[] | undefined

      @Field({type: nullable(list(nullable(TSGraphQLString)))})
      public arrayWithNullableItemField: Array<string | undefined> | undefined

      @Field({type: list(nullable(TSGraphQLString))})
      public nonnullArrayWithNullableItemField!: Array<string | undefined>
    }

    @ObjectType()
    class Query {
      @Field({type: SampleObject})
      public sampleQuery(): SampleObject {
        return {} as SampleObject
      }
    }

    const schema = compileSchema({Query})
    const {errors} = await graphql(schema, getIntrospectionQuery())
    expect(errors).toBeFalsy()
    expect(printSchema(schema)).toMatchSnapshot()
  })
})
