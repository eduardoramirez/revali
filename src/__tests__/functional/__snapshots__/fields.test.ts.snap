// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields generates a schema for valid fields 1`] = `
"type Query {
  sampleQuery: SampleObject!
}

type SampleNestedObject {
  stringField: String!
}

type SampleObject {
  implicitStringField: String!
  objectField: SampleNestedObject!
  explicitNullableStringField: String
  implicitStringArrayField: [String!]!
  nullableArrayFieldNew: [String!]
  nullableObjectArrayField: [SampleNestedObject!]
  arrayWithNullableItemField: [String]
  nonnullArrayWithNullableItemField: [String]!
}
"
`;
